docs/resources/scim_integration.md

```diff
@@ -25,13 +28,16 @@ resource "snowflake_scim_integration" "aad" {
 
 ### Required
 
-- `name` (String) Specifies the name of the SCIM integration. This name follows the rules for Object Identifiers. The name should be unique among security integrations in your account.
-- `provisioner_role` (String) Specify the SCIM role in Snowflake that owns any users and roles that are imported from the identity provider into Snowflake using SCIM.
+- `enabled` (Boolean) Specify whether the security integration is enabled.
```

Comment:
what was the behavior before? was it enabled by default or disabled by default?

Also, this is a breaking change (the old config won't work); please add this to the migration guide.

Reply:
It was disabled by default. Added to migration guide.

Reply:
The migration guide's purpose is to guide the user through the migration. It's not enough to write: `Field `enabled` is now required.` We have to be explicit about what the user has to do, what was the previous behavior, etc. In this case, if the previous behavior was disabled by default, then we should add something like: `Previously the default behavior was disabled, so please add enabled: false if you want to mimic the old one.` (or something better).

The alternative would be to consider setting a default in this case (it would be an exception to our baseline that we usually steer away from defaults). But maybe, in this case, it would be okay because it is not based on any Snowflake default underneath but would be a default in our provider. Even in this case, a short sentence for the user would be appreciated.

@sfc-gh-jmichalak @sfc-gh-jcieslak what do you think?

Reply:
I think we can describe the change more clearly in the migration guide and leave it without a default (?). I think it makes it more readable from the user's point of view to see that the migration is not abled rather than have it hidden away.

Reply:
It turned out that this field is actually optional, so I changed SDK and marked `enabled` as optional. But the behavior of this field is odd: property_default is false, but when you run a create without ENABLED, the property_value is true. When you do an UNSET, the `property_value` is false. I think we should state this in field description or introduce some custom logic to handle this case. What do you think?

Reply:
Yeah, maybe document it and later on resolve this with security integration guys or doc discuss team that would follow up with them (because the documentation is probably not including that either).

Reply:
Marked field `enabled` as required. Updated tests and added info to migration guide.